overlord/assertstate/validation_set_tracking.go

```diff
@@ -170,7 +172,7 @@ func EnforcedValidationSets(st *state.State, extraVs *asserts.ValidationSet) (*s
 
 		// if extraVs matches an already enforced validation set, then skip that one, extraVs has been added
 		// before the loop.
-		if extraVs != nil && extraVs.AccountID() == vs.AccountID && extraVs.Name() == vs.Name {
+		if skip[fmt.Sprintf("%s:%s", vs.AccountID, vs.Name)] {
```

Comment:
do we have a test where this is true multiple times in the for loop?

Reply:
Only one was covered, added a case where two extra vs replace both tracked vs.

Reply:
Only one was covered, added a case where two extra vs replace both tracked vs.